app/components/login/loginController.js

```diff
@@ -0,0 +1,24 @@
+eShopApp.controller('LoginController', function ($scope, login, $window, session) {
+    $scope.login = function (username, loginForm) {
+        if (loginForm.$valid) {
+            login.getUser(username)
+                .then(function(response) {
+
+                    if(response.data){
+                        session.login(response.data);
+                        toastr.success('Welcome ' + response.data.username);
+                        $window.location = "#/home";
+                    }
+                    else{
+                        toastr.error('Incorrect Username');
+                    }
+                })
+                .catch(function (response) {
+                    console.log(response);
+                });
+        }
+    };
+
+    $scope.usernamePattern = /^(?:[A-Za-z0-9]+)(?:[A-Za-z0-9]*)$/;
```

Comment:
We Can define Constant file that carry the patterns defined on the application

Reply:
I made a service for patterns